### DISCUSSION TITLE
Option to skip views from localhost?

### DISCUSSION THREAD

**Discussion by surjithctly** [umami-software/umami#852] [Q&A]
It would be great if Umami could skip views from localhost because there might be times when we keep developing projects even after adding Analytics.

If there is an option already, please guide me. I searched existing issues but did not find any.

**Comment by mikecao** [comment]
Either
1) modify your code to only render the tag when in production
2) or add `umami.disabled = true` to your browser localStorage

**Reply by adityatelange** [reply]
copy and paste in console (for anyone who does not want to do this manually)
```
localStorage.setItem('umami.disabled', 1)
```

**Comment by WLun001** [comment]
Possible to disable it via query params like [Panelbear](https://panelbear.com/docs/exclude-browser/)

**Comment by LeoSeyers** [comment]
> It would be great if Umami could skip views from localhost because there might be times when we keep developing projects even after adding Analytics.
> 
> If there is an option already, please guide me. I searched existing issues but did not find any.

There is the `data-domains` option (https://umami.is/docs/tracker-config) for that purpose as well

**Comment by phosphotungstic** [comment]
I test on prod (I'm just a microblog, don't shoot!)
So using `data-domains` doesn't really work for me.
I've added a PR that could add the feature to both have it temporarily or set it as a localStorage option from the URI.

**Comment by lorddarq** [comment]
Ideally you would want to configure umami to filter out based on ip as well as domain. to cover both the localhost case as well as to ensure only production uses tracking, and only the ips that are not bound to you.

What I am trying to say is that it needs a exclusion setting to help filter out ips and domains.

**Comment by ghost** [comment]
The solution for me was to complete remove the tracking script from the sources and setting an placeholder. 
When I deploy my site the placeholder will be replaced with the tracking script. 
So in the development there is no tracking and only when I deploy the tracking is 'enabled'.
